### ISSUE TITLE
Add `color-function-alias-notation`

### ISSUE THREAD

**Issue by rakleed** [stylelint/stylelint#8446]
### What is the problem you're trying to solve?

Form https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/rgb:

> [!NOTE]
> The `rgba()` functional notation is an alias for `rgb()`. They are exactly equivalent. It is recommended to use `rgb()`.

### What solution would you like to see?

Report and suggest an auto-fix to replace `rgba` with `rgb`.

**Comment by Mouvedia** [comment]
Hello @rakleed,

What you are requesting is partly covered by the `color-function-notation` rule.

**Comment by rakleed** [comment]
@Mouvedia seems like it doesn't work for my case: https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLpANhAE7wpEDmARkgBQBMArAyo86ygPQoMCkAlAG4sAXywgANCABmAS3wwAckgC2cRDAAeKgA7yAdGADOhieAiZZ5BCAyYUKdCE04YmKIcekA2lnv3HhjgAnvL4Mpg4ALSQFjLkkYFIbkhEUI6+KAC6IqYxlgBixMpIONYAVobmprDaJoi2-iCBITBhEZ4OIPglMIGO4hkBwaHhUXlxCThJUClpCJ3dLn0gOcJAA

**Comment by Mouvedia** [comment]
I understand that.

[The `rgba` function is not deprecated apparently.](https://github.com/w3c/csswg-drafts/issues/10886#issuecomment-2356573475)
Even if it was, a `function-no-deprecated` rule wouldn't be warranted for only one case.

**Comment by jeddy3** [comment]
@rakleed Thanks for the request.

I think we can consider this a false negative in [`color-function-notation`](https://stylelint.io/user-guide/rules/color-function-notation/) as it already autofixes legacy _aliases_ when commas separate the parameters.

For example, this:

```css
a {
  color: rgba(0, 0, 0, 5%);
}
```

Becomes:

```css
a {
  color: rgb(0 0 0 / 5%);
}
```

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLpANhAE7wpEDmARkgBQAMANCo80wKwCkAlANxYC+WEAxAAzAJb4YAOSQBbOIhgAPeQAcpAOjABnHcPARME8ghAZMKFOhAqcMTFB03SAbSxWrNnTgCeU-HFMHABaSGNxchCfJEckIigbDxQAXQEDcJMAMWI5JBwzACsdIwNYNX1ECy8QH38YQOCXaxB8fJgfGwZk7z8AoNDMyOicWKh4xIQWtvtOkHT+IA)

From the [spec](https://drafts.csswg.org/css-color-4/#color-syntax-legacy):

> legacy color syntax which has the following differences... non-opaque forms use a separate notation (for example [hsla()](https://drafts.csswg.org/css-color-4/#funcdef-hsla) rather than [hsl()](https://drafts.csswg.org/css-color-4/#funcdef-hsl)) and the alpha term is separated by commas (optionally preceded and/or followed by whitespace)

It should also flag and fix legacy aliases for the `"modern"` primary option when separated by whitespace. 

For example, this

```css
a {
  color: rgba(0 0 0 / 5%);
}
```

Becomes:

```css
a {
  color: rgb(0 0 0 / 5%);
}
```

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by jeddy3** [comment]
@rakleed I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new rule](https://stylelint.io/developer-guide/rules#add-a-rule) in the Developer guide.

**Comment by romainmenke** [comment]
@carlosjeurissen Do you have browser support stats and specific issues of where `rgba()` is required?

Even MDN deleted the individual pages for `rgba` and `hsla` and now just redirects to `rgb` and `hsl` because support is considered to be universal at this point.

**Comment by shrpne** [comment]
@romainmenke 

https://caniuse.com/mdn-css_types_color_rgb_alpha_parameter
looks like this is it

94.46% global support as of today

**Comment by romainmenke** [comment]
Thank you for linking that @shrpne,

Manually verified with:

```css
background-color: rgb(0, 255, 0, 1);
color: rgba(0, 255, 0, 1);
```

In Chrome 65 this renders fully lime, in Chrome 64 only text is lime colored.
I wonder if maybe MDN data merged those two functions a bit too early?

-----

@carlosjeurissen can you file a new issue to discuss that extra plugin option?

**Comment by carlosjeurissen** [comment]
@romainmenke @shrpne see https://github.com/stylelint/stylelint/issues/8746
